homeassistant/components/whirlpool/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The Whirlpool Sixth Sense integration."""
+import asyncio
+import logging
+
+import aiohttp
+from whirlpool.auth import Auth
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import AUTH_INSTANCE_KEY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Whirlpool Sixth Sense component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Whirlpool Sixth Sense from a config entry."""
+
+    auth = Auth(entry.data["username"], entry.data["password"])
+    try:
+        await auth.load_auth_file()
```

Comment:
Don't store auth in a file. Instead store it in the `entry.data`.

Reply:
Also how can loading a file cause a ClientConnectionError? And what about all other client errors?

Reply:
After loading the file, this method checks if the access token is expired. If so, it does the authentication with the backend to fetch a new token.
I'll refactor this to use entry.data.

Reply:
Actually, no need to save anything. We can always use username+password here. It also takes care of updating the SAID (appliances) list too.